File: internal/bot/command/delete_google_app.go
package command

import (
	tgbotapi "github.com/go-telegram-bot-api/telegram-bot-api/v5"
	"github.com/miti99/store-scraper-bot-go/internal/config"
	"github.com/miti99/store-scraper-bot-go/internal/repository"
)

// /delgoogle <appId> — Java DeleteGoogleAppCommand.
type DeleteGoogleAppCommand struct {
	cfg       *config.Config
	adminRepo *repository.AdminRepository
	groupRepo *repository.GroupRepository
}

func NewDeleteGoogleAppCommand(cfg *config.Config, adminRepo *repository.AdminRepository, groupRepo *repository.GroupRepository) *DeleteGoogleAppCommand {
	return &DeleteGoogleAppCommand{cfg: cfg, adminRepo: adminRepo, groupRepo: groupRepo}
}

func (c *DeleteGoogleAppCommand) Execute(msg *tgbotapi.Message, sender Sender) {
	if !authorizeGroup(msg.Chat.ID, c.adminRepo, sender) {
		return
	}
	args := splitArgs(msg.CommandArguments())
	if len(args) != 1 {
		_ = sender.SendMessage(msg.Chat.ID, "Invalid arguments")
		return
	}
	removed, err := c.groupRepo.RemoveGoogleApp(msg.Chat.ID, args[0])
	if err != nil {
		_ = sender.SendMessage(msg.Chat.ID, "Internal server error")
		return
	}
	if !removed {
		_ = sender.SendMessage(msg.Chat.ID, "Google app is not added")
		return
	}
	_ = sender.SendMessage(msg.Chat.ID, "Google app deleted successfully")
}

File: src/bot/commands/delete-google-app.js
import { authorizeGroup } from './command-utils.js';

// /delgoogle <appId>
export function createDeleteGoogleAppCommand(store) {
  return async (msg, sender, args) => {
    if (!(await authorizeGroup(msg.chat.id, store, sender))) return;
    if (args.length !== 1) {
      await sender.sendMessage(msg.chat.id, 'Invalid arguments');
      return;
    }
    const removed = await store.group.removeGoogleApp(msg.chat.id, args[0]);
    if (!removed) {
      await sender.sendMessage(msg.chat.id, 'Google app is not added');
      return;
    }
    await sender.sendMessage(msg.chat.id, 'Google app deleted successfully');
  };
}
